packages/AlertDialog/index.tsx

```diff
@@ -28,122 +29,109 @@ const ModalButton = styled.TouchableOpacity`
   align-items: center;
 `;
 
-const ModalInput = styled.TextInput<ColorProps>`
+const ModalInput = styled.TextInput`
   height: 40px;
   margin-top: 20px;
   padding: 4px;
   font-size: 14px;
-  border: 1px solid ${(props) => props.color};
+  border: 1px solid black;
 `;
 
-const ButtonText = styled.Text<ColorProps>`
+const ButtonText = styled.Text`
   font-size: 14px;
   font-weight: bold;
-  color: ${(props) => props.color};
+  color: black;
 `;
 
 type Styles = {
-  modalContainer?: StyleProp<ViewStyle>;
+  modal?: ModalProps['style'];
+  button?: StyleProp<ViewStyle>;
+  buttonText?: StyleProp<TextStyle>;
 };
 
 interface Props {
   isOpen?: boolean;
   type?: 'alert' | 'confirm' | 'prompt';
-  color?: string;
   styles?: Styles;
   backdropOpacity?: number;
   backdropPressToClose?: boolean;
   children?: React.ReactNode;
   onPress: (result: boolean | string | null) => void;
 }
 
-interface ColorProps {
-  color: string;
-}
-
 const AlertDialog = React.forwardRef(
   (
     {
       isOpen = false,
       type = 'alert',
-      color = '#000000',
       styles = {},
       backdropOpacity = 0.5,
       backdropPressToClose = true,
       children,
       onPress,
     }: Props,
-    ref: React.RefObject<unknown>,
+    ref: React.RefObject<Modal>,
   ) => {
     const [input, setInput] = useState('');
 
-    const modalStyles = {
-      modal: [
-        {
-          borderRadius: 4,
-        },
-        styles?.modalContainer,
-      ],
-    };
-
     const renderAlert = (
-      <ModalButton onPress={() => onPress(true)}>
-        <ButtonText color={color}>OK</ButtonText>
+      <ModalButton style={styles?.button} onPress={() => onPress(true)}>
+        <ButtonText style={styles?.buttonText}>OK</ButtonText>
       </ModalButton>
     );
 
     const renderConfirm = (
       <>
-        <ModalButton onPress={() => onPress(false)}>
-          <ButtonText color={color}>CANCEL</ButtonText>
+        <ModalButton style={styles?.button} onPress={() => onPress(false)}>
+          <ButtonText style={styles?.buttonText}>CANCEL</ButtonText>
         </ModalButton>
-        <ModalButton onPress={() => onPress(true)}>
-          <ButtonText color={color}>OK</ButtonText>
+        <ModalButton style={styles?.button} onPress={() => onPress(true)}>
+          <ButtonText style={styles?.buttonText}>OK</ButtonText>
         </ModalButton>
       </>
     );
 
     const renderPrompt = (
```

Comment:
I'd like to point out here that `renderAlert` is not a `function`. If you want to provide this as a function I want to ask you to change it to function `renderAlert = () => ...`.

Currently, it is a component then you should separate the component.